### PR TITLE
[Disaster Dashboard] Add units to info cards

### PR DESCRIPTION
**Pull request by juliawu**
This PR enables property value units to be displayed on the disaster dashboard.

More specifically, the following changes are made:
* Extract out the util functions in `event/app.tsx` to a dedicated utils file, that way these general functions can be reused.
* Edit `displayProps` interface to map to strings instead of numbers, to keep the unit stored.
* Remove stripping the unit in the main `disaster_event_map_utils`.
* Pass that unit to the info card so it can also be displayed.
* Update property value processing in `histogram_tile` and `top_event_tile` which expect numbers instead of a number with unit.
* Update `disaster_event_map_utils_test` to reflect changes

Before:
Screenshot 2023-06-05 at 4 16 33 PM

After:
Screenshot 2023-06-05 at 4 21 25 PM

Dashboard page itself remains unchanged:
Screenshot 2023-06-05 at 4 25 19 PM
Screenshot 2023-06-05 at 4 25 28 PM
Screenshot 2023-06-05 at 4 25 37 PM
Screenshot 2023-06-05 at 4 29 31 PM
Screenshot 2023-06-05 at 4 29 40 PM
Screenshot 2023-06-05 at 4 29 48 PM

### CODE REVIEW

**Comment by juliawu** [comment]
@shifucun @chejennifer (Reviewers): This is a large change, comments on the approach also welcome. If you think a different approach would work better, I can close this PR and implement the different approach instead.

File: static/js/utils/property_value_utils.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  if (!val) {
+    return "";
+  }
+  return formatPropertyValue(val);
+}
+
+/**
+ * Parses a lat,long pair from the first value of a property value node.
+ */
+export function parseLatLong(property: Node[]): [number, number] {
```
Comment (chejennifer) [inline static/js/utils/property_value_utils.tsx:87]: nit: don't need to export helper functions

File: static/js/utils/property_value_utils.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    unit = unit.trim();
+  }
+  const num = Number.parseFloat(val.substring(numIndex));
+  return formatNumber(num, unit);
+}
+
+/**
+ * Formats the first display value of a property value node as a number with unit.
+ */
+export function formatPropertyNodeValue(property: Node[]): string {
```
Comment (chejennifer) [inline static/js/utils/property_value_utils.tsx:76]: nit: for exported functions, in the comments, want to also include @param like: https://github.com/datacommonsorg/website/blob/master/static/js/utils/geojson_utils.ts#L25-L29

File: static/js/utils/disaster_event_map_utils.tsx [hunk 3]
```diff
@@ -262,12 +262,8 @@ function fetchEventPoints(
         if (eventTypeSpec.displayProp) {
           for (const dp of eventTypeSpec.displayProp) {
             if (dp.prop in eventData.propVals) {
-              const val = parseEventPropVal(
-                eventData.propVals[dp.prop].vals,
-                dp.unit || "",
-                reqParams
-              );
-              if (!isNaN(val)) {
+              const val = eventData.propVals[dp.prop].vals[0];
```
Comment (chejennifer) [inline static/js/utils/disaster_event_map_utils.tsx:265]: check that vals is non empty before accessing the first element? Or just use the `getValue` util function to get the value with all of this handled

File: static/js/components/tiles/histogram_tile.tsx [hunk 4]
```diff
@@ -179,7 +180,8 @@ function binData(
     let eventValue = 1;
     if (property) {
       // default to 0 if property can't be found in display props
-      eventValue = event.displayProps[property] || 0;
+      eventValue =
+        stripUnitFromPropertyValue(event.displayProps[property]) || 0;
```
Comment (chejennifer) [inline static/js/components/tiles/histogram_tile.tsx:184]: should check that event.displayProps[property] is not undefined?

**Review by chejennifer — COMMENTED** [review]
nice! this approach makes sense to me

File: static/js/utils/property_value_utils.tsx (hunk 2 above)
Comment (juliawu) [reply static/js/utils/property_value_utils.tsx:80]: Added `@param`s to each exported function

File: static/js/utils/property_value_utils.tsx (hunk 1 above)
Comment (juliawu) [reply static/js/utils/property_value_utils.tsx:92]: Keeping this function exported, it gets used here: https://github.com/datacommonsorg/website/blob/b8b597ff9a5d777f3d50b3d01cfd49509354cf1d/static/js/apps/event/app.tsx#L86

File: static/js/utils/disaster_event_map_utils.tsx (hunk 3 above)
Comment (juliawu) [reply static/js/utils/disaster_event_map_utils.tsx]: Added check for non-empty, `getValue()` expects a list of nodes as input instead.

File: static/js/components/tiles/histogram_tile.tsx (hunk 4 above)
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx]: Added check!

File: static/js/components/tiles/histogram_tile.tsx [hunk 5]
```diff
@@ -177,9 +178,12 @@ function binData(
...
     // Get value of property to aggregate
     let eventValue = 1;
-    if (property) {
-      // default to 0 if property can't be found in display props
-      eventValue = event.displayProps[property] || 0;
+    if (property && event.displayProps) {
+      // Default to 0 if property can't be found in display props
+      const propertyValueString = event.displayProps[property];
+      eventValue = propertyValueString
+        ? stripUnitFromPropertyValue(event.displayProps[property])
```
Comment (chejennifer) [inline static/js/components/tiles/histogram_tile.tsx:185]: nit: pass in propertyValueString?

File: static/js/utils/disaster_event_map_utils.tsx [hunk 6]
```diff
@@ -49,6 +49,7 @@ import {
   EventTypeSpec,
   SeverityFilter,
 } from "../types/subject_page_proto_types";
+import { getValue } from "./property_value_utils";
```
Comment (chejennifer) [inline static/js/utils/disaster_event_map_utils.tsx:52]: nit: seems unused?

**Review by chejennifer — APPROVED** [review]
thanks for the updates!

File: static/js/components/tiles/histogram_tile.tsx (hunk 5 above)
Comment (juliawu) [reply static/js/components/tiles/histogram_tile.tsx:185]: Thanks for the catch! Fixed.

File: static/js/utils/disaster_event_map_utils.tsx (hunk 6 above)
Comment (juliawu) [reply static/js/utils/disaster_event_map_utils.tsx:52]: Thanks for the catch! Removed.

**Pull request merged**